app/controllers/courses_controller.rb

```diff
@@ -0,0 +1,49 @@
+class CoursesController < ApplicationController
+  before_action :find_course
+
+  def show
+  end
+
+  def new
+    @course = @path.courses.build
+  end
+
+  def edit
+  end
+
+  def create
+    @course = @path.courses.create(course_params)
+
+    redirect_to path_path(@path)
+  end
+
+  def update
+    if @course.update(course_params)
+      redirect_to path_path(@path)
+    else
+      render 'edit'
+    end
+  end
+
+  def destroy
+    @course.destroy
+
+    redirect_to path_path(@path)
+  end
+
+  private
+
+  def course_params
+    params.require(:course).permit(:title, :description)
+  end
+
+  def find_course
+    @path = Path.find_by_title(params[:path_id])
```

Comment:
for Rails 4+, it's better to use `find_by(title: 'xxx')` instead of `find_by_title('xxx')`